src/plugins/kibana_usage_collection/server/collectors/management/telemetry_management_collector.test.ts

```diff
@@ -0,0 +1,140 @@
+    });
+  });
+
+  test('fetch() should not fail if invoked when not ready', async () => {
+    getUiSettingsClient.mockImplementationOnce(() => undefined as any);
+    await expect(collector.fetch(mockedFetchContext)).resolves.toBe(undefined);
+  });
+});
+
+describe('createCollectorFetch', () => {
+  const mockUserSettings = {
+    item1: { userValue: 'test' },
+    item2: { userValue: 123 },
+    item3: { userValue: false },
+  };
+
+  const mockIsSensitive = (key: string) => {
+    switch (key) {
+      case 'item1':
+      case 'item2':
+        return false;
+      case 'item3':
+        return true;
+      default:
+        throw new Error(`Unexpected ui setting: ${key}`);
+    }
+  };
+
+  it('returns #fetchUsageStats function', () => {
+    const getUiSettingsClient = jest.fn(() => undefined);
+    const fetchFunction = createCollectorFetch(getUiSettingsClient);
+    expect(typeof fetchFunction).toBe('function');
+  });
+
+  describe('#fetchUsageStats', () => {
+    it('returns undefined if no uiSettingsClient returned from getUiSettingsClient', async () => {
+      const getUiSettingsClient = jest.fn(() => undefined);
+      const fetchFunction = createCollectorFetch(getUiSettingsClient);
+      const result = await fetchFunction();
+      expect(result).toBe(undefined);
+      expect(getUiSettingsClient).toBeCalledTimes(1);
+    });
+
+    it('returns all user changed settings', async () => {
+      const uiSettingsClient = uiSettingsServiceMock.createClient();
+      const getUiSettingsClient = jest.fn(() => uiSettingsClient);
+      uiSettingsClient.getUserProvided.mockResolvedValue(mockUserSettings);
+      uiSettingsClient.isSensitive.mockImplementation(mockIsSensitive);
+      const fetchFunction = createCollectorFetch(getUiSettingsClient);
+      const result = await fetchFunction();
+      expect(typeof result).toBe('object');
+      expect(Object.keys(result!)).toEqual(Object.keys(mockUserSettings));
+    });
+
+    it('returns the actual values of non-sensitive settings', async () => {
+      const uiSettingsClient = uiSettingsServiceMock.createClient();
+      const getUiSettingsClient = jest.fn(() => uiSettingsClient);
+      uiSettingsClient.getUserProvided.mockResolvedValue(mockUserSettings);
+      uiSettingsClient.isSensitive.mockImplementation(mockIsSensitive);
+      const fetchFunction = createCollectorFetch(getUiSettingsClient);
+      const result = await fetchFunction();
+      expect(typeof result).toBe('object');
+      expect(result!).toMatchObject({
+        item1: 'test',
+        item2: 123,
+      });
+    });
+
+    it('returns [REDACTED] as a value for sensitive settings', async () => {
+      const uiSettingsClient = uiSettingsServiceMock.createClient();
+      const getUiSettingsClient = jest.fn(() => uiSettingsClient);
+      uiSettingsClient.getUserProvided.mockResolvedValue(mockUserSettings);
+      uiSettingsClient.isSensitive.mockImplementation(mockIsSensitive);
```

Comment:
NIT: these 4 lines are duplicated in all the tests  of this `describe` section. Could probably be extracted to `beforeEach` or a function.